### PR TITLE
`kubernetes_cluster_resource` support network_profile.ip_versions

### PR DESCRIPTION
**Pull request by heiazuo**
### create a public pull request for `kubernetes_cluster_resource` support network_profile.ip_families

Link  to https://github.com/hashicorp/terraform-provider-azurerm/issues/15708

### CODE REVIEW

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @heiazuo, this is off to a good start. Aside from the comments left in-line I also have two additional suggestions:

1. We should create an additional test and that tests dual-stack networking with the value `["IPv4", "IPv6"]`
2. In the create method could we add a check that throws an error if a user tries to specify `["IPv6"]`, the validation for the property allows it, but it isn't valid for the API.

Once the above is addressed we can take another look through this.

**Comment by heiazuo** [comment]
Thanks @stephybun, thanks for your suggestion, I made the changes as suggested

File: internal/services/containers/kubernetes_cluster_resource.go [hunk 1]
```diff
@@ -2348,6 +2377,18 @@ func expandLoadBalancerProfile(d []interface{}) *containerservice.ManagedCluster
...
+	if len(input) == 0 {
+		return nil
+	}
+	ipf := make([]containerservice.IPFamily, 0)
+	for _, data := range input {
+		ipf = append(ipf, containerservice.IPFamily(data.(string)))
+	}
+
+	return &ipf
+}
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_resource.go:2390]: Like so
```suggestion
func expandIPVersions(input []interface{}) (*[]containerservice.IPFamily, error) {
	if len(input) == 0 {
		return nil, nil
	}
	ipf := make([]containerservice.IPFamily, 0)
	for _, data := range input {
		ipf = append(ipf, containerservice.IPFamily(data.(string)))
	}
	if len(ipf) == 1 && ipf[0] == containerservice.IPFamilyIPv6 {
		return nil, fmt.Errorf("`ip_versions` must be `IPv4` or `IPv4` and `IPv6`. `IPv6` alone is not supported")
	}

	return &ipf, nil
}
```

File: internal/services/containers/kubernetes_cluster_resource.go [hunk 2]
```diff
@@ -2266,13 +2293,15 @@ func expandKubernetesClusterNetworkProfile(input []interface{}) (*containerservi
...
 	outboundType := config["outbound_type"].(string)
+	ipVersions := config["ip_versions"].([]interface{})
 
 	networkProfile := containerservice.NetworkProfile{
 		NetworkPlugin:   containerservice.NetworkPlugin(networkPlugin),
 		NetworkMode:     containerservice.NetworkMode(networkMode),
 		NetworkPolicy:   containerservice.NetworkPolicy(networkPolicy),
 		LoadBalancerSku: containerservice.LoadBalancerSku(loadBalancerSku),
 		OutboundType:    containerservice.OutboundType(outboundType),
+		IPFamilies:      expandIPVersions(ipVersions),
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_resource.go:2304]:
```suggestion
	ipVersions, err := expandIPVersions(config["ip_versions"].([]interface{}))
	if err != nil {
		return nil, err
	}

	networkProfile := containerservice.NetworkProfile{
		NetworkPlugin:   containerservice.NetworkPlugin(networkPlugin),
		NetworkMode:     containerservice.NetworkMode(networkMode),
		NetworkPolicy:   containerservice.NetworkPolicy(networkPolicy),
		LoadBalancerSku: containerservice.LoadBalancerSku(loadBalancerSku),
		OutboundType:    containerservice.OutboundType(outboundType),
		IPFamilies:      ipVersions,
	}
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for renaming the property and adding the test for dual-stack networking @heiazuo, I still have a question and suggestion regarding the check on the value.

**Review by stephybun — APPROVED** [review]
Thanks @heiazuo, LGTM 🥟

**Pull request merged**